packages/compass-schema/src/stores/reducer.ts

```diff
@@ -0,0 +1,312 @@
+  state = getInitialState(),
+  action
+) => {
+  if (isAction<AnalysisStartedAction>(action, SchemaActions.analysisStarted)) {
+    return {
+      ...state,
+      analysisState: ANALYSIS_STATE_ANALYZING,
+      errorMessage: '',
+      schema: null,
+    };
+  }
+
+  if (
+    isAction<AnalysisFinishedAction>(action, SchemaActions.analysisFinished)
+  ) {
+    return {
+      ...state,
+      analysisState: action.schema
+        ? ANALYSIS_STATE_COMPLETE
+        : ANALYSIS_STATE_INITIAL,
+      schema: action.schema,
+      resultId: resultId(),
+    };
+  }
+
+  if (isAction<AnalysisFailedAction>(action, SchemaActions.analysisFailed)) {
+    return {
+      ...state,
+      ...getErrorState(action.error),
+      resultId: resultId(),
+    };
+  }
+
+  return state;
+};
+
+function getErrorState(err: Error & { code?: number }) {
+  const errorMessage = (err && err.message) || 'Unknown error';
+  const errorCode = err && err.code;
+
+  let analysisState: AnalysisState;
+
+  if (errorCode === ERROR_CODE_MAX_TIME_MS_EXPIRED) {
+    analysisState = ANALYSIS_STATE_TIMEOUT;
+  } else {
+    analysisState = ANALYSIS_STATE_ERROR;
+  }
+
+  return { analysisState, errorMessage };
+}
+
+function resultId(): string {
+  return new UUID().toString();
+}
+
+export const handleSchemaShare = (): SchemaThunkAction<void> => {
+  return (dispatch, getState, { namespace }) => {
+    const { schema } = getState();
+    const hasSchema = schema !== null;
+    if (hasSchema) {
+      void navigator.clipboard.writeText(JSON.stringify(schema, null, '  '));
+    }
+    dispatch(_trackSchemaShared(hasSchema));
+    openToast(
+      'share-schema',
+      hasSchema
+        ? {
+            variant: 'success',
+            title: 'Schema Copied',
+            description: `The schema definition of ${namespace} has been copied to your clipboard in JSON format.`,
+            timeout: 5_000,
+          }
+        : {
+            variant: 'warning',
+            title: 'Analyze Schema First',
+            description: 'Please Analyze the Schema First from the Schema Tab.',
+            timeout: 5_000,
+          }
+    );
+  };
+};
+
+export const _trackSchemaShared = (
+  hasSchema: boolean
+): SchemaThunkAction<void> => {
+  return (dispatch, getState, { track, connectionInfoRef }) => {
+    const { schema } = getState();
+    // Use a function here to a) ensure that the calculations here
+    // are only made when telemetry is enabled and b) that errors from
+    // those calculations are caught and logged rather than displayed to
+    // users as errors from the core schema sharing logic.
+    const trackEvent = () => ({
+      has_schema: hasSchema,
+      schema_width: schema?.fields?.length ?? 0,
+      schema_depth: schema ? calculateSchemaDepth(schema) : 0,
+      geo_data: schema ? schemaContainsGeoData(schema) : false,
+    });
+    track('Schema Exported', trackEvent, connectionInfoRef.current);
+  };
+};
+
+/**
+ * @return {Object} initial schema state.
```

Comment:
Nit: if we even want to keep this helpful comment around, TS actually parses JSDoc in some cases, so it's better to drop it if you provided an actual type 🙂 

```suggestion
 * @return initial schema state.
```